.github/workflows/cwf-operator.yml

```diff
@@ -0,0 +1,75 @@
+name: cwf-operator
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+########## CWF-PRECOMMIT ##########
+
+  cwf-operator-precommit:
+
+    runs-on: ubuntu-latest
+
+    env:
+      GO111MODULE: on
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Export Magma root path
+      run: echo "MAGMA_ROOT=${PWD}" >> $GITHUB_ENV
+    - name: Run golang_before_install.sh script
+      run: ./circleci/golang_before_install.sh
+    - name: Move into cwf dir
+      run: cd ${MAGMA_ROOT}/cwf/k8s/cwf_operator
+    - name: Run go mod download with retry
+      uses: nick-invision/retry@v2
+      with:
+        command: cd ${MAGMA_ROOT}/cwf/k8s/cwf_operator && go mod download
+        timeout_minutes: 10
+    - name: Run precommit
+      run: |
+          make -C ${MAGMA_ROOT}/cwf/k8s/cwf_operator precommit
+#    - magma_slack_notify
```

Comment:
Currently Circle-CI will post a notification to Slack on failed builds on master. Because this workflow runs on both PRs and merges, you'll want to restrict the action to only trigger when those two conditions are true (refs == master && job failed).

There is the option of setting up Slack integration with the [Github app](https://github.com/integrations/slack#configuration), but from what I could see from the configuration, there's no way to configure it to notify on changes from checks.

Since there's _several_ actions for sending [Slack notifications](https://github.com/marketplace?type=actions&query=Slack), I'd suggest going with one of the top supported/starred ones: [rtCamp/action-slack-notify](https://github.com/marketplace/actions/slack-notify) or [8398a7/notify-slack](https://github.com/marketplace/actions/action-slack). From what I can see they all work essentially the same way, requiring a `SLACK_WEBHOOK` secret and some optional configuration for how messages should be formatted.

Here's an example using the `rtCamp` action:
```
- name: Slack Notification
  uses: rtCamp/action-slack-notify@v2
  env:
      SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
  if: (failure() || cancelled()) && ${{ github.ref == 'refs/heads/main' }}
 ```
 
Notifications also happen when new artifacts are created after a merge to master, which either of the slack notification actions I linked should support, though it looks like the one from 8398a7 provides a bit more flexibility on the message format.